src/Orleans.Connections.Security/Hosting/HostingExtensions.IClientBuilder.cs

```diff
@@ -48,16 +48,21 @@ public static class OrleansConnectionSecurityHostingExtensions
             X509Certificate2 certificate,
             Action<TlsOptions> configureOptions)
         {
-            if (certificate == null)
+            if (certificate is null)
             {
                 throw new ArgumentNullException(nameof(certificate));
             }
 
-            if (configureOptions == null)
+            if (configureOptions is null)
             {
                 throw new ArgumentNullException(nameof(configureOptions));
             }
 
+            if (!certificate.HasPrivateKey)
```

Comment:
It's better to put the check and throw into a helper method to avoid code duplication.

Reply:
Thanks, done